From bf5bca0e18f677463e56e342cccab93cb5604715 Mon Sep 17 00:00:00 2001
From: Jon TURNEY <jon.turney@dronecode.org.uk>
Date: Mon, 9 Jun 2014 16:11:23 -0700
Subject: [PATCH 07/13] Subject: Fix compilation when ./configure'd with
 --disable-asm

In file included from glapi_dispatch.c:90:0:
../../../src/mapi/glapi/glapitemp.h:1640:45: error: no previous prototype for 'glReadBufferNV' [-Wmissing-prototypes]
../../../src/mapi/glapi/glapitemp.h:4198:45: error: no previous prototype for 'glDrawBuffersNV' [-Wmissing-prototypes]
../../../src/mapi/glapi/glapitemp.h:6377:45: error: no previous prototype for 'glFlushMappedBufferRangeEXT' [-Wmissing-prototypes]
../../../src/mapi/glapi/glapitemp.h:6389:49: error: no previous prototype for 'glMapBufferRangeEXT' [-Wmissing-prototypes]
../../../src/mapi/glapi/glapitemp.h:6401:45: error: no previous prototype for 'glBindVertexArrayOES' [-Wmissing-prototypes]
../../../src/mapi/glapi/glapitemp.h:6413:45: error: no previous prototype for 'glDeleteVertexArraysOES' [-Wmissing-prototypes]
../../../src/mapi/glapi/glapitemp.h:6433:45: error: no previous prototype for 'glGenVertexArraysOES' [-Wmissing-prototypes]
../../../src/mapi/glapi/glapitemp.h:6445:50: error: no previous prototype for 'glIsVertexArrayOES' [-Wmissing-prototypes]

See also commits 8f3be339 (which fixed this once already), and 3ed8d42 (which
reverted the fix to fix it somewhere else [1])

[1] http://lists.freedesktop.org/archives/mesa-dev/2012-July/023899.html

Change-Id: I66b5f480eff46ec513c5ad6b9704c96dd202b15e
Signed-off-by: Jon TURNEY <jon.turney@dronecode.org.uk>
---
 src/mapi/glapi/glapi_dispatch.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/src/mapi/glapi/glapi_dispatch.c b/src/mapi/glapi/glapi_dispatch.c
index d2dd965..a64ad46 100644
--- a/src/mapi/glapi/glapi_dispatch.c
+++ b/src/mapi/glapi/glapi_dispatch.c
@@ -88,6 +88,13 @@
 #define _GLAPI_SKIP_PROTO_ENTRY_POINTS
 #endif

+#undef GL_COVERAGE_SAMPLES_NV
+#define __gl3_h_
+#define GL_EXT_multisampled_render_to_texture 1
+#define GL_EXT_texture_storage 1
+#include "GLES2/gl2platform.h"
+#include "GLES2/gl2ext.h"
+
 /* These prototypes are necessary because GLES1 library builds will create
  * dispatch functions for them.  We can't directly include GLES/gl.h because
  * it would conflict the previously-included GL/gl.h.  Since GLES1 ABI is not
--
1.9.1
